framework/src/fvkernels/FVFluxKernel.C

```diff
@@ -86,27 +88,23 @@ FVFluxKernel::onBoundary(const FaceInfo & fi) const
 bool
 FVFluxKernel::skipForBoundary(const FaceInfo & fi) const
 {
-  if (!onBoundary(fi))
-    return false;
+  // Boundaries to avoid come first, since they are always obeyed
+  if (avoidBoundary(fi))
+    return true;
 
+  // Blanket forcing on boundary
   if (_force_boundary_execution)
-  {
-    bool force = true;
-    for (const auto bnd_id : fi.boundaryIDs())
-      if (_boundaries_to_not_force.find(bnd_id) != _boundaries_to_not_force.end())
-      {
-        force = false;
-        break;
-      }
-
-    if (force)
-      return false;
-  }
+    return false;
 
+  // Selected boundaries to force
   for (const auto bnd_to_force : _boundaries_to_force)
     if (fi.boundaryIDs().find(bnd_to_force) != fi.boundaryIDs().end())
       return false;
 
+  // We're not on a boundary, so in practice we're not 'skipping'
+  if (!onBoundary(fi))
+    return false;
```

Comment:
So you want to be able to avoid internal boundaries using this new logic above (e.g. the `avoidBoundary` method)? Otherwise I would keep this up top because it is the most common case. Otherwise you are doing O(N) loops over the boundaries to force and O(log N) set finds in the inner loop for no reason

Reply:
Yeah I want to be able to skip internal boundaries. I ll move this a little higher, to skip the search over boundaries_to_force